splink/linker.py

```diff
@@ -414,34 +405,6 @@ def _random_sample_sql(
     ):
         raise NotImplementedError("Random sample sql not implemented for this linker")
 
-    @property
-    def _verify_link_only_job(self):
-        cache = self._intermediate_table_cache
-        if "__splink__df_concat_with_tf" not in cache:
```

Comment:
I think we should have a simplified version of this in either the linker or the validation.

Attempting a link only job with a single input df now results in a random error that isn't immediately obvious.

<details>
<summary>Example Code</summary>

```python
from splink.duckdb.duckdb_linker import DuckDBLinker
import pandas as pd

from tests.basic_settings import get_settings_dict

df = pd.read_csv("./tests/datasets/fake_1000_from_splink_demos.csv")

settings = get_settings_dict()
settings["link_type"] = "link_only"

linker = DuckDBLinker(
    df,
    settings
)

linker.estimate_u_using_random_sampling(target_rows=1e6)
```

</details>

Perhaps we should raise an issue and I'll add it to the easy fixes list?